src/journal/journald-context.c

```diff
@@ -784,3 +840,40 @@ void client_context_acquire_default(Server *s) {
 
         }
 }
+
+#if HAVE_PCRE2
+static bool is_log_matched(pcre2_code *regex, const char *message) {
+        int r;
+        _cleanup_(sym_pcre2_match_data_freep) pcre2_match_data *md = NULL;
+
+        assert(regex);
+
+        md = sym_pcre2_match_data_create(1, NULL);
+        if (!md)
+                return log_oom();
+
+        r = sym_pcre2_match(regex, (PCRE2_SPTR8) message, strlen(message), 0, 0, md, NULL);
+        if (r == PCRE2_ERROR_NOMATCH) {
+                return false;
+        } else if (r < 0) {
```

Comment:
Ditto for braces

Reply:
Also, we should return the actual error here from pcre2 here, similar to how it's done in pattern_compile()